### PR TITLE
Cleanup `NET6_0_OR_GREATER`

### PR DESCRIPTION
**Pull request by xtqqczze**
`NetCoreAppMinimum` is `net8.0`.

Total one occurrence of `NET6_0_OR_GREATER` in the repo.

### CODE REVIEW

**Comment by xtqqczze** [comment]
Related: https://github.com/dotnet/runtime/issues/109226

File: src/libraries/Common/src/System/Collections/Generic/EnumerableHelpers.cs [hunk 1]
```diff
@@ -29,7 +29,7 @@ internal static T[] ToArray<T>(IEnumerable<T> source, out int length)
         {
             // Copied from Array.MaxLength in System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Array.cs
             const int ArrayMaxLength = 0X7FFFFFC7;
-#if NET6_0_OR_GREATER
```
[the remark this answers is not in the document]
Comment (xtqqczze) [reply src/libraries/Common/src/System/Collections/Generic/EnumerableHelpers.cs:32]: Yes

**Review by eiriktsarpalis — APPROVED** [review]

File: src/libraries/Common/src/System/Collections/Generic/EnumerableHelpers.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (xtqqczze) [reply src/libraries/Common/src/System/Collections/Generic/EnumerableHelpers.cs:32]: > Is this the only occurrence in the entire repo?

Ahh, I think my search was filtering on `System.Private.CoreLib`

[the remark this answers is not in the document]
Comment (xtqqczze) [reply src/libraries/Common/src/System/Collections/Generic/EnumerableHelpers.cs:32]: > Is this the only occurrence in the entire repo?

Ahh, I had excluded tests from my search.

**Comment by xtqqczze** [comment]
Contributes to https://github.com/dotnet/runtime/issues/109254

**Pull request merged**